Helper/Config.php

```diff
@@ -1761,6 +1773,21 @@ public function getAllConfigSettings()
         return $boltSettings;
     }
 
+    /**
+     * Set config setting name to the given value
+     *
+     * @param string $settingName
+     * @param mixed $settingValue
+     */
+    public function setConfigSetting($settingName, $settingValue = null)
+    {
+        $this->configWriter->save(
+            self::CONFIG_SETTING_PATHS[$settingName], 
```

Comment:
I am not sure but can you consider saving only if the setting really changed? Updating setting even if with the same value can take time and trigger same events.